src/transform/src/join_implementation.rs

```diff
@@ -694,6 +837,9 @@ mod differential {
 
             let (start, mut start_key, start_characteristics) = order[0].clone();
 
+            // Count new arrangements for this choice of ordering.
+            let new_arrangements = inputs.len() + new_input_arrangements[start];
```

Comment:
Could you please add a comment here that this is relying on `optimize_orders` enumerating orders in a way that the ith order starts with input i? And maybe add a doc comment on `optimize_orders` that its return value should have this property. (I think the code so far hasn't been relying on this assumption anywhere, right?)

Reply:
The code does rely on this property: delta joins must have the first input as the first thing in the first path (or lowering to LIR fails).